test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -386,20 +634,32 @@ fi
 if [ "$newer" = true ]; then
     for i in ${!newer_release_branches[@]}; do
         [[ $((i+1)) < ${#newer_release_branches[@]} ]] && \
-        (verify_branches ${newer_release_branches[$i]} ${newer_release_branches[$((i+1))]} "row" true)
+        (verify_test_format ${newer_release_branches[$i]} ${newer_release_branches[$((i+1))]} "row" true)
+    done
+    for i in ${!test_checkpoint_release_branches[@]}; do
+        [[ $((i+1)) < ${#test_checkpoint_release_branches[@]} ]] && \
+        (verify_test_checkpoint ${test_checkpoint_release_branches[$i]} ${test_checkpoint_release_branches[$((i+1))]} "row")
     done
 fi
 
 if [ "$older" = true ]; then
     for i in ${!older_release_branches[@]}; do
         [[ $((i+1)) < ${#older_release_branches[@]} ]] && \
-        (verify_branches ${older_release_branches[$i]} ${older_release_branches[$((i+1))]} "fix row var" true)
+        (verify_test_format ${older_release_branches[$i]} ${older_release_branches[$((i+1))]} "fix row var" true)
+    done
+fi
+
+if [ "${patch_version}" = true ]; then
+    for b in ${patch_version_upgrade_downgrade_release_branches[@]}; do
+        for pv in ${pversions[@]}; do
+            (verify_test_checkpoint "$b" "$pv" "row")
+        done
     done
```

Comment:
@kommiharibabu I just realized this for loop will make all release branch latest's cross-check all picked patch versions. e.g. 
- mongodb-5.0 with mongodb-5.0.2
- mongodb-4.4 with mongodb-4.4.6
- mongodb-5.0 with mongodb-4.4.6
- mongodb-4.4 with mongodb-5.0.2

I guess the first 2 are expected, but not the last 2?

Reply:
Yes, the last 2 may not be required. I thought of having those scenarios will be good to test the direct upgrade from an older version to the latest new release and also downgrade to the latest release. Let me know your thoughts.